source/Halibut/Transport/Protocol/MessageExchangeStream.cs

```diff
@@ -465,6 +471,24 @@ TemporaryFileStream CopyStreamToFile(Guid id, long length, BinaryReader reader)
             }
             return new TemporaryFileStream(path, log);
         }
+        
+        async Task<TemporaryFileStream> CopyStreamToFileAsync(Guid id, long length, Stream stream, CancellationToken cancellationToken)
```

Comment:
The sync version uses a `BinaryReader`, and this doesn't.

From what I can see, there is nothing that the BinaryReader does that the Stream could not do.

But the fact it's there makes me wonder if there is a reason it's there that we are not aware of.

Do you think it worth looking at history and/or reaching out to find out?

Reply:
It might be because previously we had wrapped the stream in a binary stream, and we just happened to pass that around.